temporal-testing/src/main/java/io/temporal/internal/testservice/TestWorkflowService.java

```diff
@@ -222,21 +255,38 @@ private TestWorkflowService(boolean isOutOfProc, int port) {
   @Override
   public void close() {
     log.debug("Shutting down TestWorkflowService");
-    client.close();
+
+    if (outOfProcessServer != null) {
+      log.info("Shutting down out-of-process GRPC server");
+      outOfProcessServer.shutdown();
+    }
+
+    if (inProcessServer != null) {
+      log.info("Shutting down in-process GRPC server");
+      inProcessServer.shutdown();
+      client.channel.shutdown();
+    }
+
+    forkJoinPool.shutdown();
 
     try {
+      forkJoinPool.awaitTermination(1, TimeUnit.SECONDS);
+
       if (outOfProcessServer != null) {
         outOfProcessServer.shutdown();
         outOfProcessServer.awaitTermination(1, TimeUnit.SECONDS);
-      } else {
+      }
+
+      if (inProcessServer != null) {
+        inProcessServer.awaitTermination(1, TimeUnit.SECONDS);
```

Comment:
Do you mind implementing `client.awaitTermination` that would await on both stubs and channel?